pandas/tseries/resample.py

```diff
@@ -673,18 +670,15 @@ def aggregate(self, arg, *args, **kwargs):
     def _get_new_index(self):
         """ return our new index """
         ax = self.ax
-        ax_attrs = ax._get_attributes_dict()
-        ax_attrs['freq'] = self.freq
-        obj = self._selected_obj
 
         if len(ax) == 0:
-            new_index = PeriodIndex(data=[], **ax_attrs)
-            return obj.reindex(new_index)
-
-        start = ax[0].asfreq(self.freq, how=self.convention)
-        end = ax[-1].asfreq(self.freq, how='end')
+            values = []
+        else:
+            start = ax[0].asfreq(self.freq, how=self.convention)
+            end = ax[-1].asfreq(self.freq, how='end')
```

Comment:
I disagree, do this with ``.asfreq()`` rather than pad and it is clear that the snap dates
when upsampling. are either start/end depending on the convention. 

In [25]: s = pd.Series(range(2), pd.period_range('1/1/1990', periods=2, freq='M'))

In [26]: s
Out[26]: 
1990-01    0
1990-02    1
Freq: M, dtype: int64

```
In [27]: s.resample('5D', convention='start').asfreq()
Out[27]: 
1990-01-01    0.0
1990-01-06    NaN
1990-01-11    NaN
1990-01-16    NaN
1990-01-21    NaN
1990-01-26    NaN
1990-01-31    NaN
Freq: 5D, dtype: float64

In [28]: s.resample('5D', convention='end').asfreq()
Out[28]: 
1990-01-31    0.0
1990-02-05    NaN
1990-02-10    NaN
1990-02-15    NaN
1990-02-20    NaN
1990-02-25    NaN
Freq: 5D, dtype: float64
```